artemis-server/src/main/java/org/apache/activemq/artemis/core/server/Queue.java

```diff
@@ -361,6 +361,10 @@ int moveReferences(int flushLimit,
 
    int retryMessages(Filter filter) throws Exception;
 
+   default int retryMessages(Filter filter, Integer expectedHits) throws Exception {
```

Comment:
Why default method? Its been implemented in queueimpl

Reply:
there are a bunch of Fakes that don't require implementing anything.

Reply:
@michaelandrepearce I could go without the default.. but having the default here minimized the number of changes on Fakes.. etc... It was just for simplicity and minimize the number of lines changed.